modules/task/task.go

```diff
@@ -75,6 +79,10 @@ func newAsyncModule(
 	_globalBackendMu.Lock()
 	_globalBackend = backend
 	_globalBackendMu.Unlock()
+	// Register context implementation with the encoder so it can be used in async functions
+	if err := _globalBackend.Encoder().Register(context.Background()); err != nil {
```

Comment:
agreed, I don't like this either. I think the right solution would be something like extracting the values from context and passing it through backend and constructing it back into context on the receiving side. Planning to discuss this with them today.